### PR TITLE
Creation of countries list screen

### PR DESCRIPTION
**Pull request by MauricioNavaC**
# Creation of countries list screen
## Summary
Countries screen UI

* [x] Fix RecyclerView items data when scrolls down
* [x] Search View added
* [x] No data case added
* [x] Toolbar UI done
* [ ] Country flag

## Screenshots / Video
countries_screen_ui

### CODE REVIEW

File: app/src/main/java/com/globant/countriesappglb/country/hub/CountriesListAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                countryCapitalTextView.text = country.capital
+                Glide.with(countryFlagImageView.context)
+                    .load(PROGRESS_PLACEHOLDER)
+                    .into(countryFlagImageView)
+            }
+        }
+    }
+
+    companion object {
+        private const val PROGRESS_PLACEHOLDER =
```
Comment (edwinperaza) [inline app/src/main/java/com/globant/countriesappglb/country/hub/CountriesListAdapter.kt:60]: I would suggest to use a local placeholder. Lets add an asset so we can use it. What do you think?

File: app/src/main/java/com/globant/countriesappglb/country/hub/CountriesListAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.codechallenge.dickssportinggoods.models.Country
+import com.globant.countriesappglb.databinding.ItemCountryBinding
+
+class CountriesListAdapter(
+    private var countries: ArrayList<Country>,
+    private val listAction: ListAction,
+) : RecyclerView.Adapter<CountriesListAdapter.CountryViewHolder>() {
```
Comment (josecabrera13) [inline app/src/main/java/com/globant/countriesappglb/country/hub/CountriesListAdapter.kt:13]: Can we use ListAdapter and the class DiffUtil for the adapter ?

File: app/src/main/res/layout/item_country.xml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    <androidx.appcompat.widget.AppCompatImageView
+        android:id="@+id/countryFlagImageView"
+        android:layout_width="@dimen/ui_dimen_64"
+        android:layout_height="@dimen/ui_dimen_64"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintBottom_toBottomOf="parent" />
+
+    <androidx.appcompat.widget.LinearLayoutCompat
```
Comment (josecabrera13) [inline app/src/main/res/layout/item_country.xml:18]: If we use constraintLayout we should try to keep a plain layout.

File: app/src/main/java/com/globant/countriesappglb/country/hub/ListAction.kt [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package com.globant.countriesappglb.country.hub
+
+import com.codechallenge.dickssportinggoods.models.Country
+
+interface ListAction {
```
[the remark this answers is not in the document]
Comment (MauricioNavaC) [reply app/src/main/java/com/globant/countriesappglb/country/hub/ListAction.kt]: Thanks for the recommendation, it was already implemented in commit: [605657a](605657ac098c1a798803503a5c97d4838d71cb91)

**Review by edwinperaza — APPROVED** [review]
LGTM. We can resolve the TODO in a following PR.

**Pull request merged**